nestkernel/stimulating_backend.h

```diff
@@ -0,0 +1,179 @@
+/*
+ *  stimulating_backend.h
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef STIMULATING_BACKEND_H
+#define STIMULATING_BACKEND_H
+
+// C++ includes:
+#include <vector>
+
+// Includes from sli:
+#include "dictdatum.h"
+#include "name.h"
+#include "dictutils.h"
+#include "stimulating_device.h"
+
+namespace nest
+{
+
```

Comment:
Please add some developer-level documentation about the class describing the basic operations for `StimulatingBackend`s. See the corresponding [documentation of `RecordingBackend`](https://github.com/nest/nest-simulator/blob/master/nestkernel/recording_backend.h#L39-L67) for what I mean.

Reply:
I add the document based on the corresponding documentation.